src/D2L.Bmx/Okta/Models/AuthenticateResponse.cs

```diff
@@ -0,0 +1,31 @@
+using System.Text.Json.Serialization;
+namespace D2L.Bmx.Okta.Models;
+
+internal class AuthenticateResponse {
+	public DateTimeOffset ExpiresAt { get; set; }
+
+	[JsonConverter( typeof( JsonStringEnumConverter ) )]
+	public AuthenticationStatus Status { get; set; }
+}
+
+internal class AuthenticateResponseInital : AuthenticateResponse {
+	public string? StateToken { get; set; }
```

Comment:
I believe source generated json serializer should work with record's primary constructor, so we don't need the setters here, and can use records.
You can set property attributes on a primary constructor like this:
https://github.com/Brightspace/oauth2-iam-grant-libraries/blob/eb2d0b58b873cfd3f91cb8f4a90414473341c54d/src/D2L.IamGrant.Client/IamGrantLambdaResponse.cs

Reply:
Also I'm not sure if we really need the base class/inheritance structure here. Do we actually need these properties? @boarnoah

Reply:
@boarnoah I got rid of the inheritance structure here but I can add it back in if necessary

Reply:
We can add it back if we want to use a type discriminator to parse response differently (i.e https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/polymorphism)  if its asks for MFA vs without, we can get to that after we have the initial implementation working.

Reply:
Ok sounds good then. Would we change the records back into classes in that case?

Reply:
No, they just used classes in the example in the docs but polymorphic de serialization works just fine with records.